File: src/noc/core/network/NocSlaveUnit.cc
#include "noc/core/network/NocSlaveUnit.hh"
#include "sim/eventq.hh"
#include "debug/NocTiming.hh"
#include <bit>


namespace gem5 {
namespace noc {
namespace garnet {


NocSlaveUnit::NocSlaveUnit(const Params &p) : NetworkInterface(p)
{ }

// void
// NSU::init()
// {
//     // Initialization code
// }

void
NocSlaveUnit::addNode(std::vector<MessageBuffer *>& in,
                          std::vector<MessageBuffer *>& out)
{
    inNode_ptr = in;
    outNode_ptr = out;

    for (auto& it : in) {
        if (it != nullptr) {
            it->setConsumer(this, this);
        }
    }
}

void
NocSlaveUnit::print(std::ostream& out) const
{
    out << "[NocSlaveUnit " << m_id << "]";
}

}
}
}

File: src/noc/core/network/NocSlaveUnit.py
from m5.objects.ClockedObject import ClockedObject
from m5.objects.NocNetwork import NocGarnetNetworkInterface
from m5.params import *
from m5.proxy import *


class NocSlaveUnit(NocGarnetNetworkInterface):
    type = "NocSlaveUnit"
    cxx_class = "gem5::noc::garnet::NocSlaveUnit"
    cxx_header = "noc/core/network/NocSlaveUnit.hh"
    # data_width is inherited from NocGarnetNetworkInterface

    abstract = True
